app/services/invoices/calculate_fees_service.rb

```diff
@@ -138,6 +146,37 @@ def should_not_create_charge_fee?(charge, subscription)
         charge.included_in_next_subscription?(subscription)
     end
 
+    def create_fixed_charge_fees(subscription, boundaries)
+      return unless fixed_charge_boundaries_valid?(boundaries)
+
+      subscription.fixed_charges.find_each do |fixed_charge|
+        next unless should_create_fixed_charge_fee?(fixed_charge, subscription)
+
+        Fees::FixedChargeService.call!(
+          invoice:,
+          fixed_charge:,
+          subscription:,
+          boundaries:,
+          context:
+        )
+      end
+    end
+
+    # In current PR we just always create the fixed charges. In the upcoming we'll handle upgrade/downgrade/termination scenarios
+    def should_create_fixed_charge_fee?(fixed_charge, subscription)
+      # when "starting" invoice - it's only for pay_in_advance fees
+      if !fixed_charge.pay_in_advance? && subscription.invoice_subscriptions.count == 1 &&
+          subscription.invoice_subscriptions.first.subscription_starting?
```

Comment:
these complex conditions would be great to have them encapsulated in a method with a self describing naming. It is kind of hard to follow.